src/vector_index.c

```diff
@@ -95,27 +131,27 @@ int VectorQuery_EvalParams(dict *params, QueryNode *node, QueryError *status) {
       return REDISMODULE_ERR;
     }
   }
-  for (size_t i = 0; i < QueryNode_NumParams(node->vn.vq); i++) {
-    int res = VectorQuery_ParamResolve(&node->vn.vq->params[i], params, status);
+  for (size_t i = 0; i < array_len(node->vn.vq->params.params); i++) {
+    int res = VectorQuery_ParamResolve(node->vn.vq->params, i, params, status);
     if (res < 0) {
       return REDISMODULE_ERR;
     }
   }
   return REDISMODULE_OK;
 }
 
-int VectorQuery_ParamResolve(VectorQueryParam *param, dict *params, QueryError *status) {
-  if (!param->isParam) {
+int VectorQuery_ParamResolve(VectorQueryParams params, size_t ix, dict *paramsDict, QueryError *status) {
```

Comment:
why ix?

Reply:
why i :) ? short for index but I can change that